### ISSUE TITLE
Can't connect to Deepl

### ISSUE THREAD

**Issue by Gfurst** [marisukukise/japReader#59]
Unfortunately it seems to have broken down yet once again.  Also nice touch on adding it.

![Screenshot_20230926_124343](https://github.com/marisukukise/japReader/assets/12454141/fe8047db-f07b-4a98-a0d7-a29a1764144a)

It was working fine yesterday.  
Maybe unrelated or not, but sometimes it just stopped translating and wouldn't come back until restarted.  
Also, probably unrelated, but having a `%` character break japreader, for both the translation and furigana window.

**Comment by Gfurst** [comment]
Just tried it out now, while the latest one (1.9.2) wasn't connecting the previous version does work. Its probably just Deepl messing shit up.

**Comment by Ashdemai** [comment]
Getting the same warning too.

**Comment by Ashdemai** [comment]
> What about 1.9.3?

Fixes the issue.

**Comment by gutocbs** [comment]
I'm having the same problem, updating to the version 1.9.3 did not fix it for me.

**Comment by Ashdemai** [comment]
> I need a few people that it doesn't work for to do this:
> 
>     1. Go to https://www.deepl.com/translator#ja/en/ (being logged out, if you have a deepL account)
> 
>     2. Right click -> Save page as...
> 
>     3. Upload the downloaded .htm or .html file here

[DeepL Translate The world's most accurate translator.zip](https://github.com/marisukukise/japReader/files/12771325/DeepL.Translate.The.world.s.most.accurate.translator.zip)

I put it in a zip since you can't upload html

**Comment by gutocbs** [comment]
[DeepL Translate - O melhor tradutor do mundo.zip](https://github.com/marisukukise/japReader/files/12771419/DeepL.Translate.-.O.melhor.tradutor.do.mundo.zip)

**Comment by Ashdemai** [comment]
I'd like to add this anecdote: 1.9.3 works fine on my laptop which runs on Fedora but on my PC, which runs Arch even 1.9.4-pr.1 doesn't solve the problem.

**Comment by gutocbs** [comment]
With the developer tools, the problem seems to be a CORs error.

`Access to fetch at 'https://w.deepl.com/web?request_type=jsonrpc&il=pt-BR&method=getClientState' from origin 'https://www.deepl.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

It is weird that the script can't find the right class.
The line selector document.querySelector('.dl_body'); returns null when running the code, but it finds the selector in the browser.

**Comment by Gfurst** [comment]
Just a FYI, but since I had a different system installed version (1.9.1) and locally tested that one (1.9.2). I've just been using the former and its been working since then. Sometimes it gets stuck but probably another issue.

**Comment by Ashdemai** [comment]
Seems like this issue hasn't been touched in a while. I know you're doing it for free. I'm willing to put a bounty for a fix.

**Comment by Gfurst** [comment]
@marisukukise I appreciate all the work you've put in this. As for the issue, for me at least its been working without issues for the last weeks. It doesn't help that deepl seems to be always flinging their stuff around.

**Comment by Ashdemai** [comment]
> As for the issue, for me at least its been working without issues for the last weeks.

Sadly I still have this issue.

**Comment by Gfurst** [comment]
Which version are you on? Have you tried the latest test? And if so, try regressing a couple versions. Mine is working on 1.9.3

**Comment by Ashdemai** [comment]
> Which version are you on? Have you tried the latest test? And if so, try regressing a couple versions. Mine is working on 1.9.3

I'm using 1.9.3 the AUR version. I also have the latest version from Fedora repos on my laptop and it shows the same problem.

**Comment by sirbanana2** [comment]
hello, if there's anyone here looking for an alternative to the deepl feature of this program (like i was), get lunatranslator from their github page. to use the program with textractor, go into the text input tab from settings, check the clipboard setting, from the clipboard tab enable the first option, then from the embedded translation tab select deepl (there's multiple versions of it, all of them work for me). happy reading and merry christmas.
also mr developer of japreader, if you're reading this thank you for taking the mantle and continuing development from the previous project, i used both versions and they've served me well

**Comment by Ashdemai** [comment]
I wish this worked again. The "solutions" I've found are nowhere near as useful as this program did when it worked.

**Comment by Ashdemai** [comment]
> Check if v1.9.5 works

It works! Is there any way I can pay you?

**Comment by Ashdemai** [comment]
Sent some to you

**Comment by Gfurst** [comment]
I just wanted to mention, thanks for keeping it working too, I can't help right but I deeply appreciate this incredibly useful tool.
